### PR TITLE
add post-submit azure conformance

### PR DESCRIPTION
**Pull request by mjudeikis**
Add conformance tests as post submit task for azure conformance 

@kargakis ptal

### CODE REVIEW

File: ci-operator/jobs/openshift/openshift-azure/openshift-openshift-azure-master-postsubmits.yaml [hunk 1]
```diff
@@ -81,3 +81,60 @@ postsubmits:
...
       serviceAccountName: ci-operator
+  - agent: kubernetes
+    always_run: false
+    branches:
+    - master
+    - private-preview-1
+    decorate: true
+    labels:
+      artifacts: images
+    name: pull-ci-openshift-openshift-azure-master-e2e-azure-conformance
```
Comment (kargakis) [inline ci-operator/jobs/openshift/openshift-azure/openshift-openshift-azure-master-postsubmits.yaml]: Name should start with branch-ci vs pull-ci

File: ci-operator/jobs/openshift/openshift-azure/openshift-openshift-azure-master-postsubmits.yaml [hunk 2]
```diff
@@ -81,3 +81,60 @@ postsubmits:
           requests:
             cpu: 10m
       serviceAccountName: ci-operator
+  - agent: kubernetes
+    always_run: false
```
Comment (kargakis) [inline ci-operator/jobs/openshift/openshift-azure/openshift-openshift-azure-master-postsubmits.yaml]: remove

File: ci-operator/jobs/openshift/openshift-azure/openshift-openshift-azure-master-postsubmits.yaml [hunk 3]
```diff
@@ -1,5 +1,61 @@
...
+        env:
+        - name: TEST_FOCUS
+          value: Suite:openshift/conformance/parallel
+        - name: CONFIG_SPEC
+          valueFrom:
+            configMapKeyRef:
+              key: master.yaml
+              name: ci-operator-openshift-openshift-azure
+        - name: JOB_NAME_SAFE
+          value: e2e-azure-conformance
```
Comment (kargakis) [inline ci-operator/jobs/openshift/openshift-azure/openshift-openshift-azure-master-postsubmits.yaml]: @bbguimaraes does this need to be unique across presubmits and postsubmits?

Comment (bbguimaraes) [reply ci-operator/jobs/openshift/openshift-azure/openshift-openshift-azure-master-postsubmits.yaml]: It has to be unique across jobs that will act on the same input, so no.

File: ci-operator/jobs/openshift/openshift-azure/openshift-openshift-azure-master-postsubmits.yaml [hunk 4]
```diff
@@ -1,5 +1,61 @@
...
+    labels:
+      artifacts: images
+    name: branch-ci-openshift-openshift-azure-master-e2e-azure-conformance
+    skip_cloning: true
+    spec:
+      containers:
+      - args:
+        - --artifact-dir=$(ARTIFACTS)
+        - --give-pr-author-access-to-namespace=true
+        - --secret-dir=/usr/local/e2e-azure-secret
```
[the remark this answers is not in the document]
Comment (mjudeikis) [reply ci-operator/jobs/openshift/openshift-azure/openshift-openshift-azure-master-postsubmits.yaml]: so how it works in line 70?

[the remark this answers is not in the document]
Comment (bbguimaraes) [reply ci-operator/jobs/openshift/openshift-azure/openshift-openshift-azure-master-postsubmits.yaml]: That's what I was trying to say: it works for one job. This second job that's being added will try to create a secret with the same name in the same namespace and fail.

[the remark this answers is not in the document]
Comment (mjudeikis) [reply ci-operator/jobs/openshift/openshift-azure/openshift-openshift-azure-master-postsubmits.yaml]: ok, I might have a brain freeze. What I should change?

[the remark this answers is not in the document]
Comment (mjudeikis) [reply ci-operator/jobs/openshift/openshift-azure/openshift-openshift-azure-master-postsubmits.yaml]: Does this mean we have to change L47 to be unique, when L17 for the same and each template will have to have unique secret name for the same secret?

[the remark this answers is not in the document]
Comment (bbguimaraes) [reply ci-operator/jobs/openshift/openshift-azure/openshift-openshift-azure-master-postsubmits.yaml]: I suggest copying what the other templates do:
- parameterize the secret name by prefixing it with `${JOB_NAME_SAFE}-`
- mount the secret to a file with the appropriate name (`e2e-azure-conformance-post-cluster-profile` in this case)

[the remark this answers is not in the document]
Comment (stevekuznetsov) [reply ci-operator/jobs/openshift/openshift-azure/openshift-openshift-azure-master-postsubmits.yaml]: Yes, every template name (mount path etc) must be unique even if the actual underlying templates are not.

[the remark this answers is not in the document]
Comment (kargakis) [reply ci-operator/jobs/openshift/openshift-azure/openshift-openshift-azure-master-postsubmits.yaml]: Figured out this is fine and ci-operator updates the secret when it finds it there.

File: ci-operator/jobs/openshift/openshift-azure/openshift-openshift-azure-master-postsubmits.yaml [hunk 5]
```diff
@@ -1,5 +1,61 @@
...
+        env:
+        - name: TEST_FOCUS
+          value: Suite:openshift/conformance/parallel
+        - name: CONFIG_SPEC
+          valueFrom:
+            configMapKeyRef:
+              key: master.yaml
+              name: ci-operator-openshift-openshift-azure
+        - name: JOB_NAME_SAFE
+          value: e2e-azure-conformance-post
```
Comment (kargakis) [inline ci-operator/jobs/openshift/openshift-azure/openshift-openshift-azure-master-postsubmits.yaml]: I think you can revert this back to e2e-azure-conformance. It doesn't have to be unique across different types of tests (presubmits, postsubmits).

Comment (mjudeikis) [reply ci-operator/jobs/openshift/openshift-azure/openshift-openshift-azure-master-postsubmits.yaml]: reverted

File: ci-operator/jobs/openshift/openshift-azure/openshift-openshift-azure-master-postsubmits.yaml [hunk 6]
```diff
@@ -1,5 +1,61 @@
...
+        command:
+        - ci-operator
+        env:
+        - name: TEST_FOCUS
+          value: Suite:openshift/conformance/parallel
+        - name: CONFIG_SPEC
+          valueFrom:
+            configMapKeyRef:
+              key: master.yaml
+              name: ci-operator-openshift-openshift-azure
```
Comment (kargakis) [inline ci-operator/jobs/openshift/openshift-azure/openshift-openshift-azure-master-postsubmits.yaml]: needs to be updated

File: ci-operator/jobs/openshift/openshift-azure/openshift-openshift-azure-master-postsubmits.yaml [hunk 7]
```diff
@@ -1,5 +1,61 @@
...
+        - --target=[output:stable:sync]
+        command:
+        - ci-operator
+        env:
+        - name: TEST_FOCUS
+          value: Suite:openshift/conformance/parallel
+        - name: CONFIG_SPEC
+          valueFrom:
+            configMapKeyRef:
+              key: master.yaml
```
Comment (kargakis) [inline ci-operator/jobs/openshift/openshift-azure/openshift-openshift-azure-master-postsubmits.yaml]: needs to be updated

**Comment by mjudeikis** [comment]
regenerate changes are a nightmare to rebase on. reopen new one

**Pull request closed without merging**